pkg/operator/controller/controller.go

```diff
@@ -111,38 +111,57 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	}
 
 	if dns != nil {
-		// Ensure we have all the necessary scaffolding on which to place dns instances.
-		if err := r.ensureDNSNamespace(); err != nil {
-			errs = append(errs, fmt.Errorf("failed to ensure dns namespace: %v", err))
-		}
-
-		if dns.DeletionTimestamp != nil {
-			// Handle deletion.
-			if err := r.ensureOpenshiftExternalNameServiceDeleted(); err != nil {
-				errs = append(errs, fmt.Errorf("failed to delete external name for openshift service: %v", err))
-			}
-			if err := r.ensureDNSDeleted(dns); err != nil {
-				errs = append(errs, fmt.Errorf("failed to ensure deletion for dns %s: %v", dns.Name, err))
+		if dns.Spec.ManagementState != operatorv1.Unmanaged {
```

Comment:
What do you think about switching the `if` and `else` so the simpler case (`dns.Spec.ManagementState == operatorv1.Unmanaged`) comes first?  It might also make sense to use a `switch` statement (even if we only have two cases for now, `case operatorv1.Unmanaged:` and `default:`).